### ISSUE TITLE
I use gocv.HoughLines to get lines but sth ERROR 

### ISSUE THREAD

**Issue by Rogary** [hybridgroup/gocv#204]
I use gocv HoughLines  like this 
```golang
	mat := gocv.IMRead("test-data/testdata20.jpg", gocv.IMReadGrayScale)
        defer mat.Close()
	dist := gocv.NewMat()
        defer dist.Close()
	gocv.HoughLines(mat, &dist, math.Pi/180, 1, 1)
	if !dist.Empty() {
		window := gocv.NewWindow("Hello")
		window.IMShow(dist)
	}
```
Error message is : 
```
OpenCV(3.4.1) Error: Bad number of channels (Source image must have 1, 3 or 4 channels) in cvConvertImage, file /tmp/opencv-20180504-71644-brjrbc/opencv-3.4.1/modules/imgcodecs/src/utils.cpp, line 622
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: OpenCV(3.4.1) /tmp/opencv-20180504-71644-brjrbc/opencv-3.4.1/modules/imgcodecs/src/utils.cpp:622: error: (-15) Source image must have 1, 3 or 4 channels in function cvConvertImage
```
=============================================================
the same error like this 
```golang
	mat := gocv.IMRead("test-data/testdata20.jpg", gocv.IMReadGrayScale)
        defer mat.Close()
	dist := gocv.NewMat()
        defer dist.Close()
	dist1 := gocv.NewMat()
        defer dist1.Close()
	gocv.HoughLines(mat, &dist, math.Pi/180, 1, 1)
	if !dist.Empty() {
                 gocv.CvtColor(dist, &dist1, gocv.ColorGrayToBGRA)
	}
```
Error message is :  
```
OpenCV(3.4.1) Error: Assertion failed (scn == 1 && (dcn == 3 || dcn == 4)) in cvtColor, file /tmp/opencv-20180504-71644-brjrbc/opencv-3.4.1/modules/imgproc/src/color.cpp, line 11164
libc++abi.dylib: terminating with uncaught exception of type cv::Exception: OpenCV(3.4.1) /tmp/opencv-20180504-71644-brjrbc/opencv-3.4.1/modules/imgproc/src/color.cpp:11164: error: (-215) scn == 1 && (dcn == 3 || dcn == 4) in function cvtColor
```

i don't know how to use  HoughLines  result  in [hybridgroup/gocv](https://github.com/hybridgroup/gocv)

**Comment by deadprogram** [comment]
I would check that your `mat` is not empty:

```go
mat := gocv.IMRead("test-data/testdata20.jpg", gocv.IMReadGrayScale)
if mat.Empty() {
    fmt.Println("Did not load")
}
```

**Comment by Rogary** [comment]
ye im sure  it's not empty  

i find sth about this question  
HoughLines    return a  2 channels mat 
and 2 channels mat  could not be used to display

so i need covent it or just use this mat like this sample
https://github.com/hybridgroup/gocv/blob/v0.9.0/imgproc_test.go#L352

**Comment by deadprogram** [comment]
Sorry I did not understand your comment. 

If you have the wrong path then the `Mat` you tried to read in will be empty. The test passing indicates that the code itself works when called in the same way as the test.

**Comment by Rogary** [comment]
If you want Show lines , you can use HoughLinesP 
HoughLinesP return a 4channels mat , so it can used to display 

if you use 2point to draw a line  
so  x0 = channel[0]  , y0 = channel[1]    , x1 = channel[2] , y1 = channel[3]

like this 
```
        mat := gocv.IMRead("test-data/testdata20.jpg", gocv.IMReadGrayScale)
	matLines := gocv.NewMat()
	matCanny := gocv.NewMat()
        if mat.Empty() {
            fmt.Println("Did not load")
        }
	gocv.Canny(mat, &matCanny, 100, 100)
	gocv.HoughLinesP(matCanny, &matLines, 0.5, math.Pi/360, 300)

	if !mat3.Empty() {
		for index1 := 0; index1 < matLines.Rows(); index1++ {
			pt1 := image.Pt(int(matLines.GetVeciAt(index1, 0)[0]), int(matLines.GetVeciAt(index1, 0)[1]))
			pt2 := image.Pt(int(matLines.GetVeciAt(index1, 0)[2]), int(matLines.GetVeciAt(index1, 0)[3]))
			gocv.Line(&matAfter, pt1, pt2, color.RGBA{255, 0, 0, 0}, 10)
		}
	}
```

BUT  this function has so less param
i need HoughLinesP  with more Param

**Comment by deadprogram** [comment]
We just released v0.13.0 which includes this code. Thanks again, @Rogary 

I am going to close this issue now, please reopen if needed. Thank you!
